qiskit/primitives/base_estimator.py

```diff
@@ -0,0 +1,236 @@
+        psi1_H23_result = e([theta1]*2, grouping=[(0, 1), (0, 2)])
+        print(psi1_H23_result)
+
+        # calculate [ <psi2(theta2)|H2|psi2(theta2)> ]
+        psi2_H2_result = e([theta2], grouping=[(1, 1)])
+        print(psi2_H2_result)
+
+        # calculate [ <psi1(theta1)|H1|psi1(theta1)>, <psi1(theta3)|H1|psi1(theta3)> ]
+        psi1_H1_result2 = e([theta1, theta3], grouping=[(0, 0), (0, 0)])
+        print(psi1_H1_result2)
+
+        # calculate [ <psi1(theta1)|H1|psi1(theta1)>,
+        #               psi2(theta2)|H2|psi2(theta2)>,
+        #               <psi1(theta3)|H3|psi1(theta3)> ]
+        psi12_H23_result = e([theta1, theta2, theta3], grouping=[(0, 0), (1, 1), (0, 2)])
+        print(psi12_H23_result)
+
+"""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Callable, Iterable, Sequence
+
+from qiskit.circuit import QuantumCircuit, Parameter
+from qiskit.circuit.parametertable import ParameterView
+from qiskit.exceptions import QiskitError
+from qiskit.quantum_info import SparsePauliOp
+
+from .estimator_result import EstimatorResult
+
+
+@dataclass(frozen=True)
+class Group:
+    """The dataclass represents indices of circuit and observable."""
+
+    circuit_index: int
+    observable_index: int
+
+
+class BaseEstimator(ABC):
+    """Estimator base class.
+
+    Base class for Estimator that estimates expectation values of quantum circuits and observables.
+    """
+
+    def __init__(
+        self,
+        circuits: Iterable[QuantumCircuit],
+        observables: Iterable[SparsePauliOp],
+        parameters: Iterable[Iterable[Parameter]] | None = None,
+    ):
+        """
+        Args:
+            circuits (Iterable[QuantumCircuit]): quantum circuits that represents quantum states
+            observables (Iterable[SparsePauliOp]): observables
+            parameters (Iterable[Iterable[Parameter]]): parameters of quantum circuits.
+                Defaults to `[circ.parameters for circ in circuits]`
+
+        Raises:
+            QiskitError: for mismatch of circuits and parameters list.
+
+        The indexing is such that `parameters[i, j]` is the j-th formal parameter of `circuits[i]`.
+
+        Creating an instance of an Estimator, or using one in a `with` context opens a session that
+        holds resources until the instance is `close()`ed or the context is exited.
+        """
+        self._circuits = tuple(circuits)
+        self._observables = tuple(observables)
+        if parameters is None:
+            self._parameters = tuple(circ.parameters for circ in circuits)
+        else:
+            if len(parameters) != len(circuits):
+                raise QiskitError(
+                    f"Different number of parameters ({len(parameters)} and circuits ({len(circuits)}"
+                )
+            self._parameters = tuple(ParameterView(par) for par in parameters)
```

Comment:
```suggestion
            self._parameters = tuple(ParameterView(par) for par in parameters)
            if len(self._parameters) != len(self._circuits):
                raise QiskitError(
                    f"Different number of parameters ({len(self._parameters)} and circuits ({len(self._circuits)}"
                )
```

Reply:
Thank you. I understand it now. You are right.